airflow/api/common/experimental/mark_tasks.py

```diff
@@ -145,6 +145,9 @@ def set_state(task, execution_date, upstream=False, downstream=False,
                         dr.state = state
                         session.merge(dr)
 
+                if drs and commit:
```

Comment:
Why are you committing here? Commit happens below and makes sure it happens in one transaction.